File: Data Structures & Algorithms/binary-tree-diameter/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    private int result = 0;

    public int diameterOfBinaryTree(TreeNode root) {
        calculateDiameterOfBinaryTree(root);
        return result;
    }

    private int calculateDiameterOfBinaryTree(TreeNode root) {
        if (root == null) {
            return 0;
        }
        var left = calculateDiameterOfBinaryTree(root.left);
        var right = calculateDiameterOfBinaryTree(root.right);
        result = Integer.max(result, left + right);
        return 1 + Integer.max(left, right);
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        return self.countGoodNodes(root, root.val)
        

    def countGoodNodes(self, current, maxValue) -> int:
        if current is None:
            return 0
        good = 1 if current.val >= maxValue else 0
        maxValue = max(current.val, maxValue)
        left = self.countGoodNodes(current.left, maxValue)
        right = self.countGoodNodes(current.right, maxValue)
        return good + left + right
